public/js/Histogram.js

```diff
@@ -0,0 +1,95 @@
+var d3 = require('d3');
+var _ = require('lodash');
+_.mixin(require('lodash-deep'));
+
+function Histogram(sampleData, featureNames) {
+    this.sampleData = sampleData;
+    this.featureNames = featureNames;
+    this.feature = 0;
+
+    this.fullWidth = 500;
+    this.fullHeight = 400;
+
+    this.margin = {top: 20, right: 30, bottom: 20, left: 45};
+    this.width = 500 - this.margin.left - this.margin.right;
+    this.height = 400 - this.margin.top - this.margin.bottom;
+
+    this.drawHistogram(0);
+}
+
+Histogram.prototype.drawHistogram = function(feature) {
+
+    var self = this;
+    self.destroy();
+
+    var featureDist = _.deepPluck(self.sampleData,
+        ['feature_vector_std', feature]);
+
+    // define xScale and xAxis
+    var xScale = d3.scale.linear()
+        .domain([d3.min(featureDist), d3.max(featureDist)])
+        .range([0, this.width]);
+
+    var xAxis = d3.svg.axis()
+        .scale(xScale)
+        .orient('bottom');
+
+    var data = d3.layout.histogram()
+        .bins(xScale.ticks(10))
+    (featureDist);
```

Comment:
Ok this is super-groovy indentation/syntax. Should the call (?) not be aligned with the . from `.bins`?